internal/pkg/cli/task_run.go

```diff
@@ -681,6 +681,10 @@ Did you tag your secrets with the "copilot-application" and "copilot-environment
 		if err := o.displayLogStream(); err != nil {
 			return err
 		}
+		if exitCode, err := o.runner.CheckNonZeroExitCode(tasks); err != nil {
+			fmt.Println(err)
+			os.Exit(exitCode)
```

Comment:
This probably doesn't work super nicely with the current implementation of `CheckNonZeroExitCode`. It could result in a state where an `err` is printed out, while the program exits with `0`, which wouldn't make sense.

Perhaps we can change the method signature for `CheckNonZeroExitCode` to
```
func (r *ConfigRunner) CheckNonZeroExitCode(*int, err)
```
We return `nil, err` when we have an error, return `&exitcode, err`  or `&exitcode, nil` when we have the exit code.

Reply:
I also keep debating on whether we should `os.Exit` with the task error code. While I see the potential benefit it has for script users, I'm not sure if it's a faithful representation of the program execution.

Would love to see what folks think!

Reply:
One thing we can do here is to exit only when the exit code is greater than `0`

Reply:
I'm in favor of including the exit code from the task in `task run` - `docker run` [does the same thing](https://docs.docker.com/engine/reference/run/#exit-status). However, `kubectl run` does only [return the exit code if a some flags are set](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands) (`-i --restart=Never`), and I'd be fine with an approach like a flag that we could add in a separate PR. 

Either way, I think that the exit code should be set in main.go, not here. We could return a custom error type with an exit code in it or something like that.

Reply:
+1 to that exit code should be wrapped inside of the custom error

Reply:
As @dannyrandall suggested we are now displaying error message at main.go level and also exiting the process with the non zero exit code from there itself. 

Also for your concern @Lou1415926 re: `func (r *ConfigRunner) CheckNonZeroExitCode(*int, err)` function signature. So we have implemented `ExitCode()` function on `ErrExitCode` which we call in main.go directly so no need to keep passing this exit code param from bottom to up until main.go. So we have removed that int from the signature now. The new [commit](https://github.com/aws/copilot-cli/pull/3620/commits/f158d30d8675137365c334f4528f9fd18ff7df6f#diff-92e031f8d969d8ae755c21b2887e498847e62ecab3c96303dcf3696499aa48a0R23-R44) will have these changes.